src/event-handler/on.ts

```diff
@@ -11,10 +11,7 @@ function handleEventHandlers(
   webhookName: EmitterWebhookEventName | "error" | "*",
   handler: Function
 ) {
-  if (!state.hooks[webhookName]) {
-    state.hooks[webhookName] = [];
-  }
-
+  state.hooks[webhookName] ||= [];
```

Comment:
You'll want to revert this too, as it's even newer than optional chaining

Reply:
Yes, but TS transpliles it down to ES2019, for all platforms.

The problem was with optional chaining showing up in browser builds when support for it was recently introduced

Reply:
But then won't we have the same issue when browsers start supporting this syntax too?

Reply:
It's currently transpiled to 

```js
  var _a;

  (_a = state.hooks)[webhookName] || (_a[webhookName] = []);
```

using the `es2020` target from https://github.com/octokit/tsconfig.

So I think we are good?

Reply:
The current tooling creates a build optimized for bundlers and tooling, which is not optimised to be used in the browser directly. It bases these builds off of the `dist-src` builds produced by TypeScript (which is set to output ES2020), and then passes it into `rollup`, which basically creates a bundle directly from that build and **doesn't do any transpiling.**

`@pika/plugin-bundle-web` (requires `@pika/plugin-build-web`) **does the transpiling** using `babel` with the `@babel/preset-env` plugin, for which we can specify the targets that we want using the `.babelrc` file at the root of the repository. It defaults to browsers that support ESM, which [this table over at MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules#browser_support) explains

Reply:
honestly I'd probably just revert that change and leave the other one. We'll change the whole build setup anyway eventually, and then we can take the opportunity to use more modern ES features

```suggestion
  if (!state.hooks[webhookName]) {
    state.hooks[webhookName] = [];
  }

```